dali/base/dasds.cpp

```diff
@@ -8190,22 +8190,24 @@ void CCovenSDSManager::handleNotify(CSubscriberContainerBase *_subscriber, Memor
         factory->Release();
     }
 
-    Owned<CSubscriberNotifier> _notifier;
-    { CHECKEDCRITICALBLOCK(nfyTableCrit, fakeCritTimeout);
-        SubscriptionId id = subscriber->queryId();
-        CSubscriberNotifier *notifier = subscriberNotificationTable.find(id);
-        if (notifier)
-        {
-            notifier->queueChange(notifyData);
-            return;
-        }
-        else
-        {
-            _notifier.setown(new CSubscriberNotifier(subscriberNotificationTable, *subscriber, notifyData));
-            subscriberNotificationTable.replace(*_notifier);
-        }
+    CHECKEDCRITICALBLOCK(nfyTableCrit, fakeCritTimeout);
+    SubscriptionId id = subscriber->queryId();
+    CSubscriberNotifier *notifier = subscriberNotificationTable.find(id);
+    if (notifier)
+        notifier->queueChange(notifyData);
+    else
+    {
+        Owned<CSubscriberNotifier> _notifier = new CSubscriberNotifier(subscriberNotificationTable, *subscriber, notifyData);
+        subscriberNotificationTable.replace(*_notifier);
+        notifyPool->start(_notifier.getClear()); // NB: takes ownership (during init())
     }
-    notifyPool->start(_notifier.get());
+
+    /* Must clear before leaving ntyTableCrit block, so that the notifier thread owns and destroys the
+      * CSubscriberContainerBase object. It cannot be destroyed here, because this method may have been
+      * called inside the SDS lock during a node delete and do not want to call into subscriber manager
+      * as that can deadlock if processing a request already that requires SDS lock.
+      */
+    subscriber.clear();
```

Comment:
No, or I'm claim not anyway (please disprove), because the CSubscriberNotifier object is now a member of the subscriberNotificationTable.
It will be removed by the notify thread *once* it grabs the nfyTableCrit.

So it will have passed this point by then.